Code/Tools/ProjectManager/Source/ProjectUtils.cpp

```diff
@@ -461,8 +462,22 @@ namespace O3DE::ProjectManager
             if (projectDirectory.exists())
             {
                 // Check if there is an actual project here or just force it
-                if (force || PythonBindingsInterface::Get()->GetProject(path).IsSuccess())
+                AZ::Outcome<ProjectInfo> pInfo = PythonBindingsInterface::Get()->GetProject(path);
+                if (force || pInfo.IsSuccess())
                 {
+                    //determine if we have a restricted directory to worry about
+                    if (!pInfo.GetValue().m_restricted.isEmpty())
+                    {
+                        QDir restrictedDirectory(QStandardPaths::standardLocations(QStandardPaths::HomeLocation).first());
+                        restrictedDirectory.cd("O3DE");
+                        restrictedDirectory.cd("Restricted");
+                        restrictedDirectory.cd("Projects");
+                        restrictedDirectory.cd(pInfo.GetValue().m_restricted);
+                        if (!restrictedDirectory.isEmpty())
```

Comment:
Shouldn't this directory always be removed even if it contains files? The purpose of this function is to delete project files, not just empty directories.